### PR TITLE
Add a rank feature that will use the distribution key to provide a gl…

### PR DESCRIPTION
**Pull request by baldersheim**
…obally unique number.

unique = (docId << 16 | distributionKey)

@geirst PR

### CODE REVIEW

File: searchlib/src/tests/features/prod_features.cpp [hunk 1]
```diff
@@ -1563,6 +1565,26 @@ Test::testMatchCount()
...
+{
+    {
+        UniqueBlueprint bp;
+        EXPECT_TRUE(assertCreateInstance(bp, "unique"));
+        FtFeatureTest ft(_factory, "");
+        StringList params, in, out;
+        FT_SETUP_OK(bp, ft.getIndexEnv(), params, in, out.add("out"));
+        FT_DUMP_EMPTY(_factory, "unique");
+
+        EXPECT_TRUE(assertMatches(0, "x", "a", "matches(foo)"));
```
Comment (geirst) [inline searchlib/src/tests/features/prod_features.cpp:1579]: Should this line be here?

File: searchlib/src/vespa/searchlib/features/uniquefeature.cpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/features/uniquefeature.cpp:1]: Please update copyright.

File: searchlib/src/vespa/searchlib/features/uniquefeature.cpp [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+void
+UniqueBlueprint::visitDumpFeatures(const IIndexEnvironment &, IDumpFeatureVisitor &) const
+{
+}
+
+bool
+UniqueBlueprint::setup(const IIndexEnvironment & env,
+                        const ParameterList & )
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/features/uniquefeature.cpp:46]: Wrong indentation?

File: searchlib/src/vespa/searchlib/features/uniquefeature.h [hunk 4]
```diff
@@ -0,0 +1,32 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include <vespa/searchlib/fef/blueprint.h>
+
+namespace search::features {
+
+/**
+ * Implements the blueprint for the unique .
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/features/uniquefeature.h:10]: Seems to be some missing text here.

**Review by geirst — APPROVED** [review]
👍

File: searchlib/src/tests/features/prod_features.cpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/tests/features/prod_features.cpp:1579]: Thanks, leftover from copy-paste-mutate.

File: searchlib/src/vespa/searchlib/features/uniquefeature.cpp (hunk 2 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/features/uniquefeature.cpp:1]: Fixed

File: searchlib/src/vespa/searchlib/features/uniquefeature.cpp (hunk 3 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/features/uniquefeature.cpp:46]: Fixed

File: searchlib/src/vespa/searchlib/features/uniquefeature.h (hunk 4 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/features/uniquefeature.h]: Fixed

**Pull request merged**